### PR TITLE
use async core

### PR DESCRIPTION
**Pull request by r10s**
this pr shifts the android-ui using the new async-core-api - core-threads are handled in the core from now on and the ui will use an event-emmiter-object to get events.

~~the pr requires the core-submodule checked out at https://github.com/deltachat/deltachat-core-rust/pull/1356~~
~~before this can finally be merged, https://github.com/deltachat/deltachat-core-rust/pull/1356 has to be merged first.~~
EDIT: the core is ready :)

### CODE REVIEW

File: src/com/b44t/messenger/DcEvent.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+public class DcEvent {
+
+    public DcEvent(long eventCPtr) {
+        this.eventCPtr = eventCPtr;
+    }
+
+    @Override protected void finalize() throws Throwable {
+        super.finalize();
+        unrefEventCPtr();;
```
Comment (cyBerta) [inline src/com/b44t/messenger/DcEvent.java:11]: two semicolons :)

File: src/com/b44t/messenger/DcEventEmitter.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+public class DcEventEmitter {
+
+    public DcEventEmitter(long eventEmitterCPtr) {
+        this.eventEmitterCPtr = eventEmitterCPtr;
+    }
+
+    @Override protected void finalize() throws Throwable {
+        super.finalize();
+        unrefEventEmitterCPtr();;
```
Comment (cyBerta) [inline src/com/b44t/messenger/DcEventEmitter.java:11]: two semicolons :P

File: src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java [hunk 3]
```diff
@@ -103,27 +99,26 @@ public ApplicationDcContext(Context context) {
...
-      mvboxWakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "mvboxWakeLock");
-      mvboxWakeLock.setReferenceCounted(false); // if the idle-thread is killed for any reasons, it is better not to rely on reference counting
-
-      sentboxWakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "sentboxWakeLock");
-      sentboxWakeLock.setReferenceCounted(false); // if the idle-thread is killed for any reasons, it is better not to rely on reference counting
+    new Thread(() -> {
+      DcEventEmitter emitter = getEventEmitter();
+      while (true) {
+        DcEvent event = emitter.getNextEvent();
+        if (event==null) {
```
Comment (cyBerta) [inline src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java:106]: I don't understand how this condition can be met. 
getNextEvent returns a `DcEvent(getNextEventCPtr())` and this is cannot be null, or?

Comment (r10s) [reply src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java:106]: it is null if the context is unref'd, this is the new gist of graceful shutdowns

Comment (r10s) [reply src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java:106]: sorry, you're right. there is an error in the bindings.

Comment (r10s) [reply src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java:106]: fixed that, good catch! thanks!

**Pull request merged**